### ISSUE TITLE
Key placement produced too many columns

### ISSUE THREAD

**Issue by alex313962** [jeroen94704/klepcbgen#10]
If i try to create a pcb for my design it gives an error and the script quit

**Comment by jeroen94704** [comment]
The software is currently limited to generating a PCB for keyboard layouts with a maximum of 18 columns. Apparently the keyboard layout you used has more columns than that. If you think this is not correct, please share the json and I will take a look.

**Comment by alex313962** [comment]
well, yes my design have more than 18 colums.
http://www.keyboard-layout-editor.com/#/gists/643c06cd45f22283325e1aa89aaa87c5
i'm not good with py but i didn't understand why you have set a limit that low. My kb is, in fact, with 4 colums in surplus if compared with a standard 100% iso standard kb (that is also larger than 18 column i guess)

**Comment by jeroen94704** [comment]
The 18 column limit comes from the available number of inputs (25) on the atmega32u4. Detecting which key gets pressed requires grouping the keys in rows and columns. With 25 inputs, this means a theoretical maximum of 156 keys (12 or 13 rows with 13 or 12 columns), but wiring the keys of a rectangular keyboard layout into a square matrix would make the resulting schematic headache-inducing. So, as a compromise, I opted to reserve 7 input pins for rows and 18 for columns, resulting in the aforementioned limits. 

Given that this limit seems to be problematic for many people I may try a square matrix anyway and see what happens.

**Comment by phpbbireland** [comment]
My layout 17 columns (6x17) and I get the same error?

**Comment by jeroen94704** [comment]
That's odd. Can you post the json file of the layout (giving it a .txt extension, as suggested by @brianredbeard)?

**Comment by phpbbireland** [comment]
Here you go
[keyboard_layout.txt](https://github.com/user-attachments/files/17342467/keyboard_layout.txt)

**Comment by jeroen94704** [comment]
Yes, that should work, but there was an off-by-one bug in the code. I just pushed a fix for this, so you can re-download the code and try again.

**Comment by phpbbireland** [comment]
Many thanks, much appreciated...
Up and running... ;)
